src/components/NewPostImageUpload.tsx

```diff
@@ -45,12 +45,14 @@ const NewPostImageUpload = ({
 
   const getThumbnail = (uri: string): string | undefined => {
     const isYoutubeVideo = uri.match(
-      /^https?\/\/(?:www\.youtube(?:nocookie)?\.com\/|m\.youtube\.com\/|youtube\.com\/)?(?:ytscreeningroom\?vi?=|youtu\.be\/|vi?\/|user\/.+\/u\/\w{1,2}\/|embed\/|watch\?(?:.*)?vi?=|vi?=|\?(?:.*)?vi?=)([^#\n/<>"']*)/i
+      /\/\/((?:www|m)\.)?((?:youtube\.com|youtu.be))(\/(?:[\w-]+\?v=|embed\/|v\/)?)([\w]+)(\S+)?$/i
```

Comment:
I was about to 💣  this because I wrote a quick unit test and it failed both of the youtube link examples until you updated it. I know it's fixed now, but there should be a unit test for this and other pattern matches (like the Vimeo one below). If it's abstracted into a function in a library (which could be useful), it could be tested.  This will also catch accidental overwrites/typos from people's editors and such later.